### PR TITLE
gitg 3.18.0 (new formula)

### PR DESCRIPTION
**Pull request by tschoonj**

### CODE REVIEW

**Comment by tschoonj** [comment]
Not sure what went wrong with Travis-CI. Everything went smoothly on Jenkins though

**Comment by bfontaine** [comment]
```
==> Sandbox log
Jan 27 12:57:42 kernel[0]: Sandbox: Python(28614) deny(1) file-write-create /usr/local/Cellar/gobject-introspection/1.46.0_1/lib/gobject-introspection/giscanner/ast.pyc
Jan 27 12:57:42 kernel[0]: Sandbox: Python(28614) deny(1) file-write-create /usr/local/Cellar/gobject-introspection/1.46.0_1/lib/gobject-introspection/giscanner/dumper.pyc
...snip...
```

It’s trying to byte-compile code outside of its prefix. You can avoid that by using `ENV["PYTHONDONTWRITEBYTECODE"] = "1"`.

**Comment by tschoonj** [comment]
I am actually surprised to see that this throws an error. IMHO this should be allowed. During the `gitg` build process the `gobject-introspection` executables are used, which requires their python byte-compiled modules to be generated since these do not ship with the bottles.

**Comment by tschoonj** [comment]
@bfontaine @DomT4 

What do you think about this?

Thanks

**Comment by DomT4** [comment]
I'm fine with it given it passed Jenkins. Baptiste's suggestion on the ENV may be a good idea though. I know Tim has been thinking about a longer-term solution to the `.pyc` issue.

**Comment by tschoonj** [comment]
What I fail to understand is how this is an issue in the first place. AFAIK there's nothing 'naughty' or unexpected about these `.pyc` files being generated in this case. Whenever would launch one of the `gobject-introspection` executables they would be generated anyway.

**Comment by tdsmith** [comment]
It's not an issue. The .pyc files are not causing CI failures; the failure to write them is not fatal. @bfontaine, please don't suggest setting `PYTHONDONTWRITEBYTECODE`; it just adds cruft. The sandbox log messages are just a distraction; I've proposed filtering them out with a regex but I haven't implemented it yet.

File: Library/Formula/gitg.rb [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+  depends_on "libgit2-glib"
+  depends_on "gsettings-desktop-schemas"
+  depends_on "libgee"
+  depends_on "json-glib"
+  depends_on "libsecret"
+  depends_on "libpeas"
+  depends_on "gtkspell3"
+  depends_on "hicolor-icon-theme"
+  depends_on "gnome-icon-theme"
+  depends_on "python3" => :optional
```
Comment (tdsmith) [inline Library/Formula/gitg.rb]: `:python3`, please

**Comment by tdsmith** [comment]
I guess the question is: why is Gitg-1.0.gir missing from the cwd when g-ir-scanner is invoked, but only on Travis? Something must be making a bad assumption about paths?

This is the rule that generates it, I think:

```make
Gitg-1.0.gir: $(GITG_GIR_IN)
        $(SED) \
                -e 's/<include name="WebKit2" version="[34].0"\/>/<include name="WebKit2" version="$(WEBKIT_MAJOR_VERSION).0"\/>/g' \
                -e 's/Gitg_in/Gitg/g' \
                $< >$@
```

**Comment by bfontaine** [comment]
@tdsmith Thank you for the clarification.

**Comment by tschoonj** [comment]
Interestingly it fails only on 2 out of 3 Travis builds. This looks a bit like a race condition to me. `Gitg-1.0.gir` should be generated before `GitgExt-1.0.gir` as this depends on the former. There doesn't seem to be a rule in `libgitg-ext/Makefile.am` that would enforce this dependency though.

I could be wrong of course since `gobject-introspection` is a bit of a mystery to me :smile:

**Comment by tschoonj** [comment]
I changed the dependency to `:python3` by the way...

**Comment by mikemcquaid** [comment]
May be a better fit for the `homebrew-gui` tap, I wonder? I'm also wondering if having a dedicated tap for all the GTK stuff is a good call? Thoughts?

**Comment by tschoonj** [comment]
Well I initially also considered `homebrew-gui` but this package also comes with an API for plug-ins, so I ended up deciding against it.

I am not sure if a dedicated `gtk` tap would be useful. It would basically be a (small) subset of the `gui` tap, just like the `x11` tap. Also, several of the formulas that depend on `gtk+` or `gtk+3` are actually libraries that can be used to generate guis, and do not provide a standalone executable with a gui. I am not sure where these should go then.

Essentially, the problem is that with an increasing number of taps, it becomes less clear which tap a Formula should go into as it would fit into several. This is especially the case when taps are available that cover a particular category of formulas (science, games,...) as well as taps that deal with formulas that share a particular dependency (x11, gtk+, tex, ...).

**Comment by mikemcquaid** [comment]
> Essentially, the problem is that with an increasing number of taps, it becomes less clear which tap a Formula should go into as it would fit into several. This is especially the case when taps are available that cover a particular category of formulas (science, games,...) as well as taps that deal with formulas that share a particular dependency (x11, gtk+, tex, ...).

Agreed, I do wonder how to solve this problem. Part of the reason I was thinking of separating taps for some of these is so people like yourself could be given maintainer access to the taps you're interested in.

**Comment by mikemcquaid** [comment]
Thanks again @tschoonj!

**Pull request closed without merging**